### ISSUE TITLE
starting container process caused "exec: \"bundle\": executable file not found in $PATH"

### ISSUE THREAD

**Issue by DMW007** [SUSE/Portus#1554]
I used our examples in https://github.com/SUSE/Portus/tree/master/examples/compose to start Portus using docker-compose. So after a `git clone`I got this: 

> root@server /etc/docker/container/# docker-compose -f docker-compose.insecure.yml up
> Creating network "compose_default" with the default driver
> Creating compose_db_1
> Creating compose_portus_1
> Creating compose_background_1
> Creating compose_registry_1
> 
> ERROR: for background  Cannot start service background: oci runtime error: container_linux.go:247: starting container process caused "exec: \"bundle\": executable file not found in $PATH"
> ERROR: Encountered errors while bringing up the project.

After this, only the container `compose_portus_1` is running. It seems like the `bundle` command from the entrypoint is missing. What am I doing wrong? 

Here some information about my environment: 

```
docker --version
Docker version 17.05.0-ce, build 89658be

# docker-compose --version
docker-compose version 1.8.0, build unknown

# lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.3 LTS
Release:        16.04
Codename:       xenial

```

**Comment by s1shed** [comment]
I hacked around it thusly:

````diff

diff --git a/examples/compose/docker-compose.insecure.yml b/examples/compose/docker-compose.insecure.yml
index d72498fd..186ea098 100644
--- a/examples/compose/docker-compose.insecure.yml
+++ b/examples/compose/docker-compose.insecure.yml
@@ -41,6 +41,7 @@ services:
       # Theoretically not needed, but cconfig's been buggy on this...
       - CCONFIG_PREFIX=PORTUS
       - PORTUS_MACHINE_FQDN_VALUE=${MACHINE_FQDN}
+      - PATH=$PATH:/srv/Portus/bin
 
       # DB. The password for the database should definitely not be here. You are
       # probably better off with Docker Swarm secrets.
diff --git a/examples/compose/docker-compose.yml b/examples/compose/docker-compose.yml
index 323c661f..0321acbd 100644
--- a/examples/compose/docker-compose.yml
+++ b/examples/compose/docker-compose.yml
@@ -43,6 +43,7 @@ services:
       # Theoretically not needed, but cconfig's been buggy on this...
       - CCONFIG_PREFIX=PORTUS
       - PORTUS_MACHINE_FQDN_VALUE=${MACHINE_FQDN}
+      - PATH=$PATH:/srv/Portus/bin
 
       # DB. The password for the database should definitely not be here. You are
       # probably better off with Docker Swarm secrets.
```

**Comment by jeffersonsouza** [comment]
same as here

**Comment by rmamba** [comment]
I see same issue, trying to create an easy install for UNRAID server

```
container_linux.go:247: starting container process caused "exec: \"bundle\": executable file not found in $PATH"

Docker version 1.12.6, build 6b644ec

docker-compose version 1.18.0, build 8dd22a9
docker-py version: 2.6.1
CPython version: 2.7.13
OpenSSL version: OpenSSL 1.0.1t  3 May 2016

UNRAID Version 6.3.5
```

**Comment by zhufengyun** [comment]
same problem

**Comment by meons** [comment]
Same problem for me.
Based on your [example](https://github.com/SUSE/Portus/blob/master/examples/compose/docker-compose.yml), I looked into portus container which is based on the same docker image (opensuse/portus:head) that the background container and it seems that the file `/srv/Portus/bin/background.rb` is missing.

> bash-4.3# ls -alh srv/Portus/bin/
total 44K
drwxr-xr-x 1 root root  166 Dec 19 09:33 .
drwxr-xr-x 1 root root  654 Dec 19 09:33 ..
-rwxr-xr-x 1 root root  129 Aug 11 07:40 bundle
-rw-r--r-- 1 root root  716 Aug 11 07:40 check_db.rb
-rw-r--r-- 1 root root 2.2K Aug 11 07:40 client.rb
-rwxr-xr-x 1 root root  978 Aug 11 07:40 crono
-rw-r--r-- 1 root root 4.4K Aug 11 07:40 generate_test_data.rb
-rwxr-xr-x 1 root root  300 Aug 11 07:40 rails
-rwxr-xr-x 1 root root  245 Aug 11 07:40 rake
-rw-r--r-- 1 root root  717 Aug 11 07:40 security.rb
-rwxr-xr-x 1 root root  804 Aug 11 07:40 setup
-rwxr-xr-x 1 root root  556 Aug 11 07:40 spring

So with a missing file in entrypoint it's normal that the container does not start.
`entrypoint: bundle exec rails runner /srv/Portus/bin/background.rb`

Could you check that ?

**Comment by vitoravelino** [comment]
I might be wrong but I guess this issue is also related to https://github.com/SUSE/Portus/issues/1418. `opensuse/portus:head` is broken for a while. @mssola has been focused trying to fix all the issues around it.

**Comment by meons** [comment]
Mmm I get the same error with `opensuse/portus:latest`. It seems that is broken too.

**Comment by mssola** [comment]
This is a mix of two problems:

1. The already known #1418 which is really close to be fixed.
2. A regression fixed in #1562.

Once #1418 is gone, this one should go away. Marking as a duplicate to keep track.

**Comment by mssola** [comment]
Fixed! See my comments on #1418.
